src/main/java/UserInterface.java

```diff
@@ -0,0 +1,223 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+
+public class UserInterface {
+    public static final String COMMAND_EXIT = "bye";
+    public static final String COMMAND_LIST = "list";
+    public static final String COMMAND_MARK = "mark";
+    public static final String COMMAND_UNMARK = "unmark";
+    public static final String COMMAND_TODO = "todo";
+    public static final String COMMAND_DEADLINE = "deadline";
+    public static final String COMMAND_EVENT = "event";
+    private final Scanner uiScan;
+    private final ArrayList<Task> tasks;
+
+    public UserInterface(Scanner mainScan) {
+        this.uiScan = mainScan;
+        this.tasks = new ArrayList<>();
+    }
+
+    public void start() {
+        printGreeting();
+        loopCommandInput();
+    }
+
+    /**
+     * Reads commands from stdin and executes them using commandExec(commandInput)
+     * until the user inputs COMMAND_EXIT, upon which the function returns.
+     */
+    public void loopCommandInput() {
+        String commandInput;
+        do {
+            commandInput = uiScan.nextLine();
+            executeCommand(commandInput);
+        } while (!commandInput.equals(COMMAND_EXIT));
+    }
+
+    /**
+     * Checks the String inputted by the user and executes the appropriate command
+     * using a switch statement.
+     * @param nextLine The command to be executed.
+     */
+    private void executeCommand(String nextLine) {
+        try {
+            ArrayList<String> pieces = new ArrayList<>(Arrays.asList(nextLine.split(" ")));
+            switch (pieces.get(0)) {
```

Comment:
Can save pieces.get(0) to a variable with a descriptive variable name to descript what pieces.get(0) represent to improve code quality. Same for pieces.get(1) further below in the code.

Reply:
Thank you. I will implement this.